tests/ashley/test_forum_search.py

```diff
@@ -0,0 +1,401 @@
+        # see: apps/forum_conversation/abstract_models.py#L314
+
+        So in the end, searching for a topic subject will return the first post as result.
+        """
+        # First, let's secure Django Machina's overriding mechanism
+        topic = TopicFactory(subject="5f3gh8ka1")
+        forum = topic.forum
+        self.assertEqual(topic.subject, "5f3gh8ka1")
+
+        # When we create the first post for this topic...
+        PostFactory(subject="497jk1sav", topic=topic)
+        # The topic subject should be overriden
+        topic.refresh_from_db()
+        self.assertEqual(topic.subject, "497jk1sav")
+
+        PostFactory(topic=topic)
+        # Creating a second post does not change the topic subject again
+        topic.refresh_from_db()
+        self.assertEqual(topic.subject, "497jk1sav")
+
+        # A third post in the same forum with a different topic
+        PostFactory(topic__forum=forum)
+
+        user = UserFactory()
+        assign_perm("can_read_forum", user, forum)
+
+        # Index the post in Elasticsearch
+        call_command("rebuild_index", interactive=False)
+
+        self.client.force_login(user)
+        response = self.client.get("/forum/search/?q=497jk1sav")
+
+        self.assertContains(
+            response, "Your search has returned <b>1</b> result", html=True
+        )
+        self.assertContains(
+            response,
+            (
+                f'<a href="/forum/forum/{forum.slug:s}-{forum.id:d}/topic/{topic.slug:s}-'
+                f'{topic.id:d}/" class="topic-name-link">{topic.subject:s}</a>'
+            ),
+            html=True,
+        )
+
+    def test_forum_search_post_failure(self):
+        """Searching for a word not present in post, topic or username should return no result."""
+        post = PostFactory(text="a5g3g6k75", subject="497jk1sav")
+        user = UserFactory()
+        assign_perm("can_read_forum", user, post.topic.forum)
+
+        # Index the post in Elasticsearch
+        call_command("rebuild_index", interactive=False)
+
+        self.client.force_login(user)
+        response = self.client.get("/forum/search/?q=blabla")
+
+        self.assertContains(
+            response, "Your search has returned <b>0</b> results", html=True
+        )
+
+    def test_forum_search_only_topics(self):
+        """
+        If the "search only topics" checkbox is ticked, matching content elsewhere should not
+        return any result.
+        """
+        post1 = PostFactory(subject="497jk1sav", text="497jk1sav")
+        forum = post1.topic.forum
+        post2 = PostFactory(topic__subject="497jk1sav", topic__forum=forum)
+
+        user = UserFactory()
+        assign_perm("can_read_forum", user, forum)
+
+        # Index the post in Elasticsearch
+        call_command("rebuild_index", interactive=False)
+
+        self.client.force_login(user)
+
+        # Searching without the "search_topics" flag returns post1
```

Comment:
I had trouble understanding this test.

```suggestion
        # Searching without the "search_topics" flag returns post1 because it contains
        # the search string in its text body.
```

Reply:
Accepted :+1: